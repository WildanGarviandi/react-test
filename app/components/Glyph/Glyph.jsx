import React from 'react';

import PropTypes from 'prop-types';
import classNaming from 'classnames';

import styles from './glyph.scss';

function Glyph({ className, name }) {
  const stylename = classNaming(className, styles.glyphicon, styles[`glyphicon-${name}`]);
  return <span className={stylename} />;
}

/* eslint-disable */
Glyph.propTypes = {
  className: PropTypes.any,
  name: PropTypes.any,
};
/* eslint-enable */

Glyph.defaultProps = {
  className: '',
  name: '',
};

export default Glyph;
