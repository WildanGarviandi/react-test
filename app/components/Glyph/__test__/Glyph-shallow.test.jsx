// Glyph.react-test.js
import React from 'react';
import { shallow } from 'enzyme';
import enzymeSerializer from 'enzyme-to-json/serializer';

import Glyph from '../Glyph';

expect.addSnapshotSerializer(enzymeSerializer);

jest.mock('classnames', () => {
  const data = () => { };
  return data;
});

jest.mock('./glyph.scss', () => {
  const data = {
    glyphicon: {},
  };
  return data;
});

describe('Glph Shallow Component', () => {
  test('should have matched snapshot', () => {
    const result = shallow(<Glyph className={''} name={''} />);
    expect(result).toMatchSnapshot();
  });
});
