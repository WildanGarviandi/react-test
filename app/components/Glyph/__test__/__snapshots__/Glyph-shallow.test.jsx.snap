// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Glph Shallow Component should have matched snapshot 1`] = `<span />`;
